examples/api_call_evaluation.py

```diff
@@ -0,0 +1,289 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    delete:
+      description: deletes a single pet based on the ID supplied
+      operationId: deletePet
+      parameters:
+        - name: id
+          in: path
+          description: ID of pet to delete
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '204':
+          description: pet deleted
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+components:
+  schemas:
+    Pet:
+      allOf:
+        - $ref: '#/components/schemas/NewPet'
+        - type: object
+          required:
+            - id
+          properties:
+            id:
+              type: integer
+              format: int64
+    NewPet:
+      type: object
+      required:
+        - name
+      properties:
+        name:
+          type: string
+        tag:
+          type: string
+    Error:
+      type: object
+      required:
+        - code
+        - message
+      properties:
+        code:
+          type: integer
+          format: int32
+        message:
+          type: string
+"""
+
+
+class CurlStrToListOfKeyValuePairs(FieldOperator):
+    """This class takes a string query api and splits it into a list of key value components.
+
+    These components can then be checked individually (e.g ignoring order)"
+    For example:
+
+    curl -X GET -H "Content-Type: application/json" 'https://petstore.swagger.io/v2/pets?tags=dogs&limit=5'
+
+    becomes
+
+    [('url', 'curl -X GET -H "Content-Type: application/json" https://petstore.swagger.io/v2/pets'), ('tags', 'dogs'), ('limit', '5')]
+
+    """
+
+    def process_value(self, text: str) -> List[Tuple[str, str]]:
+        text = text.replace("%20", " ")
+        text = text.replace("'", "")
+        # text=re.sub(r"&pageSize=\d+", "", text)
+        splits = text.split("?")
+        if len(splits) == 1:
+            return [("url", text)]
+        result = []
+        (url, params) = splits
+        result.append(("url", url))
+        params_splits = params.split("&")
+        for param in params_splits:
+            key_value_splits = param.split("=")
+            if len(key_value_splits) != 2:
+                print(f"Unable to parse key value pair from string {param} in {text}")
+                continue
+            (key, value) = key_value_splits
+            value_splits = value.split(",")
+            if len(value_splits) == 1:
+                result.append((f"{key}", f"{value}"))
+
+        return result
+
+
+template = InputOutputTemplate(
+    instruction=f"Generate the API query corresponding to the user request based on the following api specification. Answer only as a CURL command, without any explanations. \n{api_spec}.",
```

Comment:
Why the api spec is not part of the task?
```suggestion
    instruction=f"Generate the API query corresponding to the user request based on the following api specification. Answer only as a CURL command, without any explanations. \n{api_spec}.",
```

Reply:
I could do it, and then I'd place it in every instance.  I think I'll do it for clarity.

Reply:
Done